### PR TITLE
Handle floating point depth images

### PR DESCRIPTION
**Pull request by phyrog**
Adds support for handling floating point depth images. They are converted to MONO8 images before being saved.

### CODE REVIEW

**Comment by phyrog** [comment]
I'm beginning to think that converting to MONO16 might be a better option (higher depth resolution and easier to convert to pointclouds [depth_image_proc seems to want uint16 images]). That would mean to reimplement convertScaleAbs, since it always converts to 8bit images.

**Comment by phyrog** [comment]
I created a separate branch with the 16bit mono images (https://github.com/phyrog/semrec/tree/16bitmono), that saves all images as PNG, which might not be desirable with rgb images.

File: src/plugins/imagecapturer/CImageCapturer.cpp [hunk 1]
```diff
@@ -114,6 +119,10 @@ namespace semrec {
       try {
 	if(sensor_msgs::image_encodings::isColor(m_imgReceived.encoding)) {
 	  cv_ptr = cv_bridge::toCvCopy(m_imgReceived, sensor_msgs::image_encodings::BGR8);
+  } else if(isDepthImage(m_imgReceived.encoding)) {
```
Comment (fairlight1337) [inline src/plugins/imagecapturer/CImageCapturer.cpp]: Thanks for the pull request. Looks good. If you fix the indentation, I'll merge it in.

**Comment by fairlight1337** [comment]
Hm, good point about the PNGs. Saving the RGB images as JPGs is fine, as the JPG artifacts don't affect us. This might be a different story for the depth images.

I think I'll add a switch in the `semrec' interface, allowing to save either as JPG or PNG, depending on what the client wants. Then the RGB images can be saved as JPG, while the depth images can be saved as PNG.

File: src/plugins/imagecapturer/CImageCapturer.cpp (hunk 1 above)
Comment (phyrog) [inline src/plugins/imagecapturer/CImageCapturer.cpp]: In which direction should I fix it? Spaces and tabs are mixed all over the file.

Comment (fairlight1337) [inline src/plugins/imagecapturer/CImageCapturer.cpp]: I guess such that it looks the same on GitHub. I try to keep the indentation consistent, but if your settings don't match the ones reflected on the repo, I'll fix them later.

I'll just merge it in.

**Comment by phyrog** [comment]
To convert to 16 bit images instead of 8bit, see this [diff](https://github.com/phyrog/semrec/commit/fb02839bf3dd38ecaf7b0024fa391445b8b41087.diff). It's possible to change this right now, since the image is automatically converted to 8 bit when saving as jpg.

**Pull request merged**